demosplan/DemosPlanCoreBundle/Logic/Statement/StatementCopier.php

```diff
@@ -519,7 +519,13 @@ public function copyStatementObjectWithinProcedureWithRelatedFiles(
         }
         // persist to get an ID for the FileContainer copying below
         $this->getDoctrine()->getManager()->persist($newStatement);
-        $this->statementService->addFilesToCopiedStatement($newStatement, $statement->getId());
+        if ([] !== $statement->getFiles()) {
+            $this->statementService->addFilesToCopiedStatement($newStatement, $statement->getId());
+        }
+
+        // We do have to flush the new copied statement here if the original statement has no FileContainers otherwise
+        // the new copied statement is already flushed while copying FileContainers in the previous method 'addFilesToCopiedStatement'.
+        $this->getDoctrine()->getManager()->flush();
```

Comment:
This would avoid a double flush...

// We do have to flush the new copied statement here if the original statement has no FileContainers otherwise
// the new copied statement is already flushed while copying FileContainers in the previous method 'addFilesToCopiedStatement'
if (0 === count($statement->getFiles())) {
    $this->getDoctrine()->getManager()->flush();
}

Reply:
thx done here https://github.com/demos-europe/demosplan-core/pull/3862/commits/c03677c7dc3c84833d9dba95fade1095f3282ae9, differently but double flush is avoided

Reply:
Guys I don't get it. Double flush isn't a thing at all, afaik.
The flush of doctrine just writes everything in the database what was persisted before. After all of this is written, doctrine cleared this "persisted storage". An additional flush will not find anything to persist.
This additional logic, which is used here is misleading, I would prefer to avoid it.

Reply:
we want to avoid double flush here but this was not the problem. We do flush copied statement once while copying file container which is part of the copy process. The problem was that we don't flush anywhere copied statement if they don't have any filecontainer